### ISSUE TITLE
SpaceEscape sample TransformationBendWorld shader broken

### ISSUE THREAD

**Issue by Basewq** [stride3d/stride#1963]
**Release Type**: Official Release

**Version**: 4.0.1.1400

**Platform(s)**: All

**Describe the bug**
`DeformFactorX` & `DeformFactorY` in `TransformationBendWorld.sdsl` gets reset to zero

**To Reproduce**
1. Run the SpaceEscape sample.

**Expected behavior**
`DeformFactorX` & `DeformFactorY` should retain their default values as written in the shader.

**Additional context**
Bug introduced in #765 caused by
https://github.com/vvvv/stride/blob/7fa2275c3f4d392bb9837d46029f46d45f8c03cb/sources/engine/Stride.Graphics/Effects/Effect.cs#L310-L318

**Workaround**
In `BendFogRenderFeature.cs` go to method `PrepareEffectPermutations` and add the lines
```
// Generate shader permuatations
renderEffect.EffectValidator.ValidateParameter(GameParameters.EnableBend, renderMesh.Mesh.Parameters.Get(GameParameters.EnableBend));
// <----- START OF THE WORKAROUND
if (renderMesh.Mesh.Parameters.Get(GameParameters.EnableBend))
{
    var eval1 = renderMesh.Mesh.Parameters.Get(TransformationBendWorldKeys.DeformFactorX);
    var eval2 = renderMesh.Mesh.Parameters.Get(TransformationBendWorldKeys.DeformFactorY);
}
// <----- END OF THE WORKAROUND
renderEffect.EffectValidator.ValidateParameter(GameParameters.EnableFog, renderMesh.Mesh.Parameters.Get(GameParameters.EnableFog));
renderEffect.EffectValidator.ValidateParameter(GameParameters.EnableOnflyTextureUVChange, renderMesh.Mesh.Parameters.Get(GameParameters.EnableOnflyTextureUVChange));
```

This will immediately evaluate the default values preventing the shader using zero'd values.

**Comment by Basewq** [comment]
@azeno Can you explain a bit about this code?
https://github.com/vvvv/stride/blob/7fa2275c3f4d392bb9837d46029f46d45f8c03cb/sources/engine/Stride.Graphics/Effects/Effect.cs#L310-L318

I am considering fixing this by changing it to
```
private static ParameterKey FindOrCreateValueKey<T>(ref EffectValueDescription binding) where T : struct
{
    var name = binding.KeyInfo.KeyName;
    var existingKey = ParameterKeys.FindByName(name) as ValueParameterKey<T>;
    var key = existingKey ?? ParameterKeys.NewValue<T>(name: name);
    // Update the default value with the one from the shader
    if (existingKey is null && binding.DefaultValue is T defaultValue)
        key.DefaultValueMetadataT.DefaultValue = defaultValue;
    return key;
}
```

However my understanding of your pull is that you want to override the existing key's default value?

**Comment by azeno** [comment]
I answered in the PR of yours.

**Comment by Basewq** [comment]
Fixed by #1990
